File: src/main/java/com/silverboxsoft/dynamodbtool/fxmodel/TableNameCondType.java
package com.silverboxsoft.dynamodbtool.fxmodel;

import java.util.ArrayList;
import java.util.List;

public enum TableNameCondType {
	PARTIAL_MATCH("Partial match"), HEAD_MATCH("Head match"), TAIL_MATCH("Tail match");

	final private String name;

	private TableNameCondType(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public static List<String> getTitleList() {
		List<String> retList = new ArrayList<>();
		for (TableNameCondType type : TableNameCondType.values()) {
			retList.add(type.getName());
		}
		return retList;
	}

	public static TableNameCondType getByName(String name) {
		for (TableNameCondType type : TableNameCondType.values()) {
			if (type.getName().equals(name)) {
				return type;
			}
			;
		}
		return null;
	}
}

File: src/main/java/com/silverboxsoft/dynamodbtool/fxmodel/TableNameCondType.kt
package com.silverboxsoft.dynamodbtool.fxmodel

import java.util.ArrayList

enum class TableNameCondType(val condition_name: String) {
    PARTIAL_MATCH("Partial match"), HEAD_MATCH("Head match"), TAIL_MATCH("Tail match");

    companion object {
        val titleList: List<String>
            get() {
                val retList: MutableList<String> = ArrayList()
                for (type in values()) {
                    retList.add(type.condition_name)
                }
                return retList
            }

        fun getByName(condition_name: String): TableNameCondType {
            for (type in values()) {
                if (type.condition_name == condition_name) {
                    return type
                }
            }
            return PARTIAL_MATCH
        }
    }
}